engines/router/missionctl/handlers/http_handler_test.go

```diff
@@ -224,6 +255,97 @@ func TestHTTPServiceBadRequest(t *testing.T) {
 	}
 }
 
+// TestLogRequestSummary checks that copyResponseToLogChannel is called with each piece of
+// info to be logged and finally, logTuringRouterRequestSummary is called.
+func TestLogRequestSummary(t *testing.T) {
+	logUtilsAllComponents := &testLogUtils{}
+	logUtilsBadEnricher := &testLogUtils{}
+
+	// Define check for goroutine called
+	checkLogRequestSummaryCalled := func(t *testing.T, l *testLogUtils) {
+		timeout := time.After(3 * time.Second)           // Wait for a maximum of 3 mins
+		ticker := time.NewTicker(500 * time.Millisecond) // Check every 500ms
+
+	wait:
+		for {
+			select {
+			case <-timeout:
+				t.Log("logTuringRouterRequestSummary not called")
+				t.Fail()
+			case <-ticker.C:
+				if atomic.LoadInt32(&l.logTuringRouterRequestSummaryCalls) == 1 {
+					break wait
+				}
+			}
+		}
+	}
+
+	tests := map[string]struct {
+		mc       missionctl.MissionControl
+		logutils *testLogUtils
+		checks   func()
+	}{
+		"log all components": {
+			mc:       &MockMissionControl{BaseMockMissionControl: *createTestBaseMissionControl()},
+			logutils: logUtilsAllComponents,
+			checks: func() {
+				logUtilsAllComponents.AssertCalled(t, "copyResponseToLogChannel", "enricher",
+					`{"value":"Init:Enrich"}`, "")
+				logUtilsAllComponents.AssertCalled(t, "copyResponseToLogChannel", "router",
+					`{"value":"Init:Enrich:Route"}`, "")
+				logUtilsAllComponents.AssertCalled(t, "copyResponseToLogChannel", "ensembler",
+					`{"value":"Init:Enrich:Route:Ensemble"}`, "")
+				checkLogRequestSummaryCalled(t, logUtilsAllComponents)
+			},
+		},
+		"log enricher error": {
```

Comment:
Should we also add tests for errors in other components like Ensembler?

Reply:
Ideally, yes. But I think the way we deal with the error response should generally be the same across each component. Since there is some asynchronous behaviour being tested, I didn't want to add repetitive tests for this.